### PR TITLE
Cnvnator: a new package.

### PR DESCRIPTION
**Pull request by weijianwen**

### CODE REVIEW

File: var/spack/repos/builtin/packages/cnvnator/package.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    depends_on('bzip2')
+    depends_on('curl')
+    depends_on('lzma')
+    depends_on('zlib')
+
+    def edit(self, spec, prefix):
+        makefile = FileFilter('Makefile')
+        # Replace CXX with CXXFLAGS
+        makefile.filter('CXX.*=.*',
+                        'CXXFLAGS = -O3 -std=c++11 \
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/cnvnator/package.py]: Will these flags work for every compiler? It's probably better to use `self.compiler.cxx11_flag`, which should be defined for every compiler that supports it.

File: var/spack/repos/builtin/packages/cnvnator/package.py [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+                        -DCNVNATOR_VERSION=\\"$(VERSION)\\" $(OMPFLAGS)')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/cnvnator/package.py]: Will this string contain a ton of spaces from the indentation? I think you can add closing and starting quotes and Python will know to combine it into a single string. Also, adding a `r` before the string makes it raw, and means that you no longer need to escape backslashes.

Comment (adamjstewart) [reply var/spack/repos/builtin/packages/cnvnator/package.py]: Also, the `-fopenmp` flag used in Makefile is compiler specific. You should replace it with `self.compiler.openmp_flag`.

File: var/spack/repos/builtin/packages/cnvnator/package.py [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+                        'override LIBS += -lz -lbz2 -lcurl -llzma')
+
+    def build(self, spec, prefix):
+        make('ROOTSYS={0}'.format(spec['root'].prefix),
+             'SAMINC={0}'.format(spec['samtools'].prefix.include),
+             'SAMDIR={0}'.format(spec['samtools'].prefix.lib),
+             'HTSDIR={0}'.format(spec['htslib'].prefix.lib))
+
+    def install(self, spec, prefix):
+        mkdir = which('mkdir')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/cnvnator/package.py]: This line isn't needed, Spack already has a `mkdir` function which wraps around `os.mkdir`.

**Comment by alalazo** [comment]
@weijianwen Can you please update this one too and ping me back?

**Comment by weijianwen** [comment]
@alalazo I've updated this package's copyright header which should be ready to merge.

**Pull request merged**